src/main/java/com/hillel/items_exchange/controller/AdvertisementController.java

```diff
@@ -160,15 +160,13 @@ public AdvertisementDto updateAdvertisement(@Validated(Exist.class) @RequestBody
             @ApiResponse(code = 200, message = "OK"),
             @ApiResponse(code = 400, message = "BAD REQUEST"),
             @ApiResponse(code = 403, message = "FORBIDDEN")})
-    public ResponseEntity<HttpStatus> deleteAdvertisement(@Valid @RequestBody AdvertisementDto dto, Principal principal)
+    public ResponseEntity<HttpStatus> deleteAdvertisement(@Valid @RequestBody Long advertisementId, Principal principal)
             throws IllegalOperationException {
 
         User owner = getUser(principal.getName());
-        validateAdvertisementOwner(dto, owner);
-        Optional<AdvertisementDto> byId = advertisementService.findDtoById(dto.getId());
-
-        if (byId.isPresent() && byId.get().equals(dto)) {
-            advertisementService.remove(dto.getId());
+        validateAdvertisementOwner(advertisementId, owner);
+        if (advertisementService.existById(advertisementId)) {
```

Comment:
Why do we need that? In line above we checked that it exists.

Reply:
Removed.